### PR TITLE
Documentation Improvements

### PR DESCRIPTION
**Pull request by vivekmig**
This fixes all docstring formatting to avoid unnecessary indents / spacing in website documentation. The list of tutorials is also updated appropriately, and additional details are added to feature permutation documentation.

### CODE REVIEW

File: captum/attr/_core/saliency.py [hunk 1]
```diff
@@ -38,70 +38,70 @@ def attribute(
...
+                        - A list of tuples with length equal to the number of
+                          examples in inputs (dim 0), and each tuple containing
+                          #output_dims - 1 elements. Each tuple is applied as the
+                          target for the corresponding example.
+
+                        Default: None
+            abs (bool, optional): Returns absolute value of gradients if set
+                        to True, otherwise returns the (signed) gradients if
+                        False.
+                        Defalut: True
```
Comment (edward-io) [inline captum/attr/_core/saliency.py]: since you're here, `Default`

Comment (vivekmig) [reply captum/attr/_core/saliency.py]: Whoops, thanks!

File: captum/attr/_core/feature_ablation.py [hunk 2]
```diff
@@ -60,132 +60,139 @@ def attribute(
...
+                        The attributions with respect to each input feature.
+                        If the forward function returns
+                        a scalar value per example, attributions will be
+                        the same size as the provided inputs, with each value
+                        providing the attribution of the corresponding input index.
+                        If the forward function returns a scalar per batch, then
+                        attribution tensor(s) will have first dimension 1 and
+                        the remaining dimensions will match the input.
+                        If a single tensor is provided as inputs, a single tensor is
+                        returned. If a tuple is provided for inputs, a tuple of
```
Comment (NarineK) [inline captum/attr/_core/feature_ablation.py]: nit: if a tuple of tensors is provided

File: captum/attr/_core/feature_permutation.py [hunk 3]
```diff
@@ -24,27 +24,47 @@ def _permute_feature(x: Tensor, feature_mask: Tensor) -> Tensor:
 
 class FeaturePermutation(FeatureAblation):
     r"""
-    This attribution method essentially implements the permutation feature
-    importance algorithm, as described here:
-    https://christophm.github.io/interpretable-ml-book/feature-importance.html
+    A perturbation based approach to compute attribution, which
+    takes each input feature, permutes the feature values within a batch,
+    and computes the difference in output between the original and shuffled.
```
Comment (NarineK) [inline captum/attr/_core/feature_permutation.py]: nit: maybe, `... and computes the difference between original and shuffled outputs for given batch. Those differences signify feature importance for the permuted features ` ?

File: captum/attr/_core/feature_permutation.py [hunk 4]
```diff
@@ -24,27 +24,47 @@ def _permute_feature(x: Tensor, feature_mask: Tensor) -> Tensor:
...
+    each input tensor is taken as a feature and shuffled independently. Passing
+    a feature mask, allows grouping features to be shuffled together.
+    Each input scalar in the group will be given the same attribution value
+    equal to the change in target as a result of shuffling the entire feature
+    group.
+
+    The forward function can either return a scalar per example, or a single
+    scalar for the full batch. If a single scalar is returned for the batch,
+    `perturbations_per_eval` must be 1, and the returned attributions will have
+    first dimension 1, corresponding to feature importance across all
```
Comment (NarineK) [inline captum/attr/_core/feature_permutation.py]: nit: what is the features are structured multi-dimensionally ? Maybe we don't want to say `first dimension 1` ?

File: captum/attr/_core/feature_permutation.py [hunk 5]
```diff
@@ -53,11 +73,11 @@ def __init__(self, forward_func: Callable, perm_func: Callable = _permute_featur
...
             forward_func (callable): The forward function of the model or
                 any modification of it
-            perm_func (callable): A function that accepts a batch of inputs and
-                a feature mask, and "permutes" the feature across the batch.
-                NOTE: one obviously does not have to perform a permutation.
-                See `_permute_feature` as an example on how to implement
-                your own permutation function.
+            perm_func (callable, optional): A function that accepts a batch of
+                inputs and a feature mask, and "permutes" the feature across
+                the batch. This defaults to a function which applies a random
```
Comment (NarineK) [inline captum/attr/_core/feature_permutation.py]: nit : `.. the features using feature mask across the batch  ... `

File: captum/attr/_core/feature_permutation.py [hunk 6]
```diff
@@ -169,6 +189,58 @@ def attribute(  # type: ignore
...
+                        The attributions with respect to each input feature.
+                        If the forward function returns
+                        a scalar value per example, attributions will be
+                        the same size as the provided inputs, with each value
+                        providing the attribution of the corresponding input index.
+                        If the forward function returns a scalar per batch, then
+                        attribution tensor(s) will have first dimension 1 and
+                        the remaining dimensions will match the input.
+                        If a single tensor is provided as inputs, a single tensor is
+                        returned. If a tuple is provided for inputs, a tuple of
```
Comment (NarineK) [inline captum/attr/_core/feature_permutation.py]: nit: as a tuple of tensors is provided ...

File: captum/attr/_core/feature_permutation.py [hunk 7]
```diff
@@ -169,6 +189,58 @@ def attribute(  # type: ignore
...
+            >>> net = SimpleClassifier()
+            >>> # Generating random input with size 10 x 4 x 4
+            >>> input = torch.randn(10, 4, 4)
+            >>> # Defining FeaturePermutation interpreter
+            >>> feature_perm = FeaturePermutation(net)
+            >>> # Computes permutation attribution, shuffling each of the 16
+            >>> # scalar input independently.
+            >>> attr = feature_perm.attribute(input, target=1)
+
+            >>> # Alternatively, we may want to ablate features in groups, e.g.
```
Comment (NarineK) [inline captum/attr/_core/feature_permutation.py]: nit: ` we may want to permute features in groups`

Comment (vivekmig) [reply captum/attr/_core/feature_permutation.py]: Good catch, thanks!

File: captum/attr/_core/feature_permutation.py (hunk 4 above)
Comment (vivekmig) [reply captum/attr/_core/feature_permutation.py]: I think it still would have first dimension 1? It's just that the attributions would be the size of one input example.

**Pull request closed without merging**